tests/integration/helpers.bash

```diff
@@ -620,3 +621,35 @@ function requires_kernel() {
 		skip "requires kernel $1"
 	fi
 }
+
+function requires_idmap_fs() {
+	local fs
+	fs=$1
+
+	stderr=$($FS_IDMAP "$fs" 2>&1 >/dev/null)
+	ret=$?
+
+	if [ $ret -ne 0 ]; then
+		case $stderr in
+		*invalid\ argument)
+			skip "$fs underlying file system does not support ID map mounts"
+			;;
+		*operation\ not\ permitted)
+			name=$(grep 'PRETTY_NAME' /etc/os-release | cut -d'=' -f2)
+			if [ "$name" == '"CentOS Stream 9"' ]; then
+				# centos kernel 5.14.0-200 contains the following snippet in
+				# can_idmap_mount():
+				# /*
+				#  * So far, there are serious concerns about the safety of idmaps.
+				#  * This mechanism requires further upstream review.
+				#  */
```

Comment:
> Could you add the URL of this snippet

The nonsensical merge request and commit is:
https://gitlab.com/redhat/centos-stream/src/kernel/centos-stream-9/-/merge_requests/131
https://gitlab.com/redhat/centos-stream/src/kernel/centos-stream-9/-/commit/c93c728360fd10a0dc65

Reply:
One should also note that it's 2023. So that feature has been in mainline for 3 years now without alleged security issues and with support for at least xfs, ext4, btrfs, overlayfs, f2fs, squashfs, tmpfs, ntfs3, fat, erofs, and more to come. With a dedicated tree as part of my general VFS tree and @sforshee as co-maintainer.